src/Microsoft.DotNet.Cli.Utils/FileAccessRetryer.cs

```diff
@@ -47,5 +48,34 @@ public static class FileAccessRetrier
                 }
             }
         }
+
+        /// <summary>
+        /// Run Directory.Move and File.Move in Windows has a chance to get IOException with HResult 0x80070005. But this error is transient.
+        /// </summary>
+        internal static void RetryOnMoveAccessFailure(Action action)
```

Comment:
Since this method no longer performs a move, could this be named `RetryOnAccessDenied`?

Reply:
This is wrap FIle.Move and Directory.Move., _file.system.File.Move.  I don't know a easier way to enforce that, I can only put that in name. Is there a better way to convey this idea?

It could be dangerous If we use this on Directory.Copy. If the copy is partial, some file could be duplicated